### ISSUE TITLE
CI failure with recent ruby head

### ISSUE THREAD

**Issue by knu** [ruby/digest#31]
https://github.com/ruby/digest/runs/3892800575?check_suite_focus=true
```
▶ Run bundle exec rake check
gem install pkg/digest-3.1.0.pre2.gem
ERROR:  Error installing pkg/digest-3.1.0.pre2.gem:
	ERROR: Failed to build gem native extension.

    No such file or directory @ dir_s_mkdir - /home/runner/.rubies/ruby-head/lib/ruby/gems/3.1.0/gems/digest-3.1.0.pre2/ext/digest/bubblebabble/.gem.20211014-1755-c7jag6

Gem files will remain installed in /home/runner/.rubies/ruby-head/lib/ruby/gems/3.1.0/gems/digest-3.1.0.pre2 for inspection.
Results logged to /home/runner/.rubies/ruby-head/lib/ruby/gems/3.1.0/extensions/x86_64-linux/3.1.0/digest-3.1.0.pre2/gem_make.out
rake aborted!
Command failed with status (1): [gem install pkg/digest-3.1.0.pre2.gem...]
/home/runner/work/digest/digest/Rakefile:46:in `block (2 levels) in <top (required)>'
/home/runner/work/digest/digest/Rakefile:39:in `block in <top (required)>'
/home/runner/work/digest/digest/vendor/bundle/ruby/3.1.0/gems/rake-13.0.6/exe/rake:27:in `<top (required)>'
/home/runner/.rubies/ruby-head/bin/bundle:23:in `load'
/home/runner/.rubies/ruby-head/bin/bundle:23:in `<main>'
Tasks: TOP => check
(See full trace by running task with --trace)
Error: Process completed with exit code 1.
```

**Comment by knu** [comment]
@deivid-rodriguez It is still happening and I have no idea how to fix this. 😥https://github.com/ruby/digest/actions/runs/1618751011

**Comment by hsbt** [comment]
@knu It's the limitation native extension of the default gems. So, We can't install the same version of them like `3.1.0.pre3` and it. Maybe, We can install it when we bump version to `3.1.0`.

Therefore, We always keep the higher version of ruby/digest rather than ruby/ruby's one.

**Comment by deivid-rodriguez** [comment]
I was able to reproduce it, but I'm no sure why it's happening either :(

**Comment by deivid-rodriguez** [comment]
Oh, indeed, it's exactly what @hsbt pointed out. `3.1.0.pre3` is already a default gem in ruby 3.1, so you can't reinstall it as a default gem. If you bump the final version or the pre version, it should be ok. We should at least print a better error here.

**Comment by knu** [comment]
Ah, that explains it, thanks!

**Comment by knu** [comment]
Just released the gem version 3.1.0 and imported it to ruby.  Thank you for the cooperation!

**Comment by deivid-rodriguez** [comment]
No problem!
